[openebs/litmus] (test): Include a litmus book to perform application node name replacement (via ansible ec2 shutdown module).

mittachaitu opened the pull request:
Signed-off-by: mittachaitu <sai.chaithanya@mayadata.io>

<!--  Thanks for sending a pull request!  Here are some tips for you -->

**What this PR does / why we need it**:
- Include a litmus book to perform application node replacement. (by shutting down the application node)
- This test requires AWS kops cluster and by shutting down the node kops will create another node(with a different name).
- Pre-chaos and post chaos will verify the application pod status by verifying the liveness pod status.
**Test logs:**:
```
ansible-playbook 2.7.4
  config file = None
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible-playbook
  python version = 2.7.15rc1 (default, Nov 12 2018, 14:31:15) [GCC 7.3.0]
No config file found; using defaults
/etc/ansible/hosts did not meet host_list requirements, check plugin documentation if this is unexpected
/etc/ansible/hosts did not meet script requirements, check plugin documentation if this is unexpected

PLAYBOOK: test.yml *************************************************************
1 plays in ./percona/chaos/app_node_failure_ups_diff_name/test.yml
 [WARNING]: Found variable using reserved name: action


PLAY [localhost] ***************************************************************

TASK [Gathering Facts] *********************************************************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:2
ok: [127.0.0.1]
META: ran handlers

TASK [include_tasks] ***********************************************************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:13
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [include_tasks] ***********************************************************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:17
included: /common/utils/create_testname.yml for 127.0.0.1

TASK [Record test instance/run ID] *********************************************
task path: /common/utils/create_testname.yml:3
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Construct testname appended with runID] **********************************
task path: /common/utils/create_testname.yml:7
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [include_tasks] ***********************************************************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:19
included: /chaoslib/aws_chaos/prerequisites_aws.yml for 127.0.0.1

TASK [Installing pip] **********************************************************
task path: /chaoslib/aws_chaos/prerequisites_aws.yml:2
changed: [127.0.0.1] => {"changed": true, "cmd": "apt install python-pip", "delta": "0:00:01.405323", "end": "2018-12-05 08:24:27.321882", "rc": 0, "start": "2018-12-05 08:24:25.916559", "stderr": "\nWARNING: apt does not have a stable CLI interface. Use with caution in scripts.", "stderr_lines": ["", "WARNING: apt does not have a stable CLI interface. Use with caution in scripts."], "stdout": "Reading package lists...\nBuilding dependency tree...\nReading state information...\npython-pip is already the newest version (9.0.1-2.3~ubuntu1).\n0 upgraded, 0 newly installed, 0 to remove and 16 not upgraded.", "stdout_lines": ["Reading package lists...", "Building dependency tree...", "Reading state information...", "python-pip is already the newest version (9.0.1-2.3~ubuntu1).", "0 upgraded, 0 newly installed, 0 to remove and 16 not upgraded."]}

TASK [Installing boto] *********************************************************
task path: /chaoslib/aws_chaos/prerequisites_aws.yml:7
changed: [127.0.0.1] => {"changed": true, "cmd": "pip install boto", "delta": "0:00:01.844309", "end": "2018-12-05 08:24:29.300764", "rc": 0, "start": "2018-12-05 08:24:27.456455", "stderr": "", "stderr_lines": [], "stdout": "Collecting boto\n  Downloading https://files.pythonhosted.org/packages/23/10/c0b78c27298029e4454a472a1919bde20cb182dab1662cec7f2ca1dcc523/boto-2.49.0-py2.py3-none-any.whl (1.4MB)\nInstalling collected packages: boto\nSuccessfully installed boto-2.49.0", "stdout_lines": ["Collecting boto", "  Downloading https://files.pythonhosted.org/packages/23/10/c0b78c27298029e4454a472a1919bde20cb182dab1662cec7f2ca1dcc523/boto-2.49.0-py2.py3-none-any.whl (1.4MB)", "Installing collected packages: boto", "Successfully installed boto-2.49.0"]}

TASK [Check the status of nodes and have count of nodes] ***********************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:22
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get nodes --no-headers | grep -v 'master'", "delta": "0:00:00.490638", "end": "2018-12-05 08:24:29.925141", "rc": 0, "start": "2018-12-05 08:24:29.434503", "stderr": "", "stderr_lines": [], "stdout": "ip-10-0-1-196.us-west-1.compute.internal   Ready   node     3h    v1.11.0\nip-10-0-1-197.us-west-1.compute.internal   Ready   node     30m   v1.11.0\nip-10-0-1-223.us-west-1.compute.internal   Ready   node     18m   v1.11.0", "stdout_lines": ["ip-10-0-1-196.us-west-1.compute.internal   Ready   node     3h    v1.11.0", "ip-10-0-1-197.us-west-1.compute.internal   Ready   node     30m   v1.11.0", "ip-10-0-1-223.us-west-1.compute.internal   Ready   node     18m   v1.11.0"]}

TASK [include_tasks] ***********************************************************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:29
included: /common/utils/update_litmus_result_resource.yml for 127.0.0.1

TASK [Generate the litmus result CR to reflect SOT (Start of Test)] ************
task path: /common/utils/update_litmus_result_resource.yml:3
changed: [127.0.0.1] => {"changed": true, "checksum": "e909735046d5c6ee8373199726aa1bce8b2c254c", "dest": "./litmus-result.yaml", "gid": 0, "group": "root", "md5sum": "f3cc97d53b74529366406c555ecc059b", "mode": "0644", "owner": "root", "size": 465, "src": "/root/.ansible/tmp/ansible-tmp-1543998270.05-107062844560661/source", "state": "file", "uid": 0}

TASK [Analyze the cr yaml] *****************************************************
task path: /common/utils/update_litmus_result_resource.yml:14
changed: [127.0.0.1] => {"changed": true, "cmd": "cat litmus-result.yaml", "delta": "0:00:00.287629", "end": "2018-12-05 08:24:30.838701", "rc": 0, "start": "2018-12-05 08:24:30.551072", "stderr": "", "stderr_lines": [], "stdout": "---\napiVersion: litmus.io/v1alpha1\nkind: LitmusResult\nmetadata:\n\n  # name of the litmus testcase\n  name: node-shutdown-up-with-diff-name \nspec:\n\n  # holds information on the testcase\n  testMetadata:\n    app: percona \n    chaostype: node-shutdown-up-diff-name \n\n  # holds the state of testcase,  manually updated by json merge patch\n  # result is the useful value today, but anticipate phase use in future \n  testStatus: \n    phase: in-progress  \n    result: none ", "stdout_lines": ["---", "apiVersion: litmus.io/v1alpha1", "kind: LitmusResult", "metadata:", "", "  # name of the litmus testcase", "  name: node-shutdown-up-with-diff-name ", "spec:", "", "  # holds information on the testcase", "  testMetadata:", "    app: percona ", "    chaostype: node-shutdown-up-diff-name ", "", "  # holds the state of testcase,  manually updated by json merge patch", "  # result is the useful value today, but anticipate phase use in future ", "  testStatus: ", "    phase: in-progress  ", "    result: none "]}

TASK [Apply the litmus result CR] **********************************************
task path: /common/utils/update_litmus_result_resource.yml:17
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl apply -f litmus-result.yaml", "delta": "0:00:00.448031", "end": "2018-12-05 08:24:31.424967", "failed_when_result": false, "rc": 0, "start": "2018-12-05 08:24:30.976936", "stderr": "", "stderr_lines": [], "stdout": "litmusresult.litmus.io/node-shutdown-up-with-diff-name configured", "stdout_lines": ["litmusresult.litmus.io/node-shutdown-up-with-diff-name configured"]}

TASK [Generate the litmus result CR to reflect EOT (End of Test)] **************
task path: /common/utils/update_litmus_result_resource.yml:27
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Analyze the cr yaml] *****************************************************
task path: /common/utils/update_litmus_result_resource.yml:38
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Apply the litmus result CR] **********************************************
task path: /common/utils/update_litmus_result_resource.yml:41
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Display the app information passed via the test job] *********************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:36
ok: [127.0.0.1] => {
    "msg": [
        "The application info is as follows:", 
        "Namespace    : app-percona-ns", 
        "Label        : name=percona"
    ]
}

TASK [include_tasks] ***********************************************************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:44
included: /common/utils/status_app_pod.yml for 127.0.0.1

TASK [Checking Percona pod is in running state] ********************************
task path: /common/utils/status_app_pod.yml:2
changed: [127.0.0.1] => {"attempts": 1, "changed": true, "cmd": "kubectl get pods -n app-percona-ns -o jsonpath='{.items[?(@.metadata.labels.name==\"percona\")].status.phase}'", "delta": "0:00:00.356346", "end": "2018-12-05 08:24:32.139700", "rc": 0, "start": "2018-12-05 08:24:31.783354", "stderr": "", "stderr_lines": [], "stdout": "Running", "stdout_lines": ["Running"]}

TASK [Get application pod name] ************************************************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:54
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get po -n app-percona-ns -l name=percona --no-headers -o=custom-columns=NAME:\".metadata.name\"", "delta": "0:00:00.356654", "end": "2018-12-05 08:24:32.642383", "rc": 0, "start": "2018-12-05 08:24:32.285729", "stderr": "", "stderr_lines": [], "stdout": "percona-6966c6db86-8gxk4", "stdout_lines": ["percona-6966c6db86-8gxk4"]}

TASK [Generate some random name for database creation] *************************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:62
changed: [127.0.0.1] => {"changed": true, "cmd": "echo $(mktemp) | cut -d '.' -f 2", "delta": "0:00:00.279384", "end": "2018-12-05 08:24:33.057051", "rc": 0, "start": "2018-12-05 08:24:32.777667", "stderr": "", "stderr_lines": [], "stdout": "VKT5yG2OtF", "stdout_lines": ["VKT5yG2OtF"]}

TASK [Create test load in the mysql database] **********************************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:70
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Get Application pod Node to perform chaos] *******************************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:82
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get pod percona-6966c6db86-8gxk4 -n app-percona-ns --no-headers -o custom-columns=:spec.nodeName", "delta": "0:00:00.352246", "end": "2018-12-05 08:24:33.588562", "rc": 0, "start": "2018-12-05 08:24:33.236316", "stderr": "", "stderr_lines": [], "stdout": "ip-10-0-1-197.us-west-1.compute.internal", "stdout_lines": ["ip-10-0-1-197.us-west-1.compute.internal"]}

TASK [Get the AWS instance Id of APP NOde] *************************************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:91
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get node ip-10-0-1-197.us-west-1.compute.internal -o jsonpath={.spec.providerID} | cut -d '/' -f 5", "delta": "0:00:00.355376", "end": "2018-12-05 08:24:34.079103", "rc": 0, "start": "2018-12-05 08:24:33.723727", "stderr": "", "stderr_lines": [], "stdout": "i-09644e5558538b7d8", "stdout_lines": ["i-09644e5558538b7d8"]}

TASK [Get region from app node] ************************************************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:97
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get nodes ip-10-0-1-197.us-west-1.compute.internal -o jsonpath='{.metadata.labels.failure-domain\\.beta\\.kubernetes\\.io\\/region}'", "delta": "0:00:00.356482", "end": "2018-12-05 08:24:34.571390", "rc": 0, "start": "2018-12-05 08:24:34.214908", "stderr": "", "stderr_lines": [], "stdout": "us-west-1", "stdout_lines": ["us-west-1"]}

TASK [set_fact] ****************************************************************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:103
ok: [127.0.0.1] => {"ansible_facts": {"instance_id": "i-09644e5558538b7d8", "region": "us-west-1"}, "changed": false}

TASK [include_tasks] ***********************************************************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:108
included: /chaoslib/aws_chaos/chaosutil_aws.yml for 127.0.0.1

TASK [Stop the AWS ec2 instances] **********************************************
task path: /chaoslib/aws_chaos/chaosutil_aws.yml:2
fatal: [127.0.0.1]: FAILED! => {"changed": false, "msg": "wait for instances running timeout on Wed Dec  5 08:29:38 2018"}
...ignoring

TASK [include_tasks] ***********************************************************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:111
included: /common/utils/check_no_of_nodes.yml for 127.0.0.1

TASK [Waiting for application pod reschedule (evict)] **************************
task path: /common/utils/check_no_of_nodes.yml:4
ok: [127.0.0.1] => {"changed": false, "elapsed": 50, "path": null, "port": null, "search_regex": null, "state": "started"}

TASK [Checking the status of chaos injected node] ******************************
task path: /common/utils/check_no_of_nodes.yml:8
changed: [127.0.0.1] => {"attempts": 1, "changed": true, "cmd": "kubectl get nodes --no-headers | grep -vi 'master'", "delta": "0:00:00.361597", "end": "2018-12-05 08:30:29.059198", "rc": 0, "start": "2018-12-05 08:30:28.697601", "stderr": "", "stderr_lines": [], "stdout": "ip-10-0-1-196.us-west-1.compute.internal   Ready   node     3h    v1.11.0\nip-10-0-1-223.us-west-1.compute.internal   Ready   node     24m   v1.11.0\nip-10-0-1-93.us-west-1.compute.internal    Ready   node     2m    v1.11.0", "stdout_lines": ["ip-10-0-1-196.us-west-1.compute.internal   Ready   node     3h    v1.11.0", "ip-10-0-1-223.us-west-1.compute.internal   Ready   node     24m   v1.11.0", "ip-10-0-1-93.us-west-1.compute.internal    Ready   node     2m    v1.11.0"]}

TASK [Checking the no.of nodes] ************************************************
task path: /common/utils/check_no_of_nodes.yml:22
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [include_tasks] ***********************************************************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:116
included: /common/utils/status_app_pod.yml for 127.0.0.1

TASK [Checking Percona post chaos pod is in running state] *********************
task path: /common/utils/status_app_pod.yml:2
FAILED - RETRYING: Checking Percona post chaos pod is in running state (30 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (29 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (28 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (27 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (26 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (25 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (24 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (23 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (22 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (21 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (20 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (19 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (18 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (17 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (16 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (15 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (14 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (13 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (12 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (11 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (10 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (9 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (8 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (7 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (6 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (5 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (4 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (3 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (2 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (1 retries left).
fatal: [127.0.0.1]: FAILED! => {"attempts": 30, "changed": true, "cmd": "kubectl get pods -n app-percona-ns -o jsonpath='{.items[?(@.metadata.labels.name==\"percona\")].status.phase}'", "delta": "0:00:00.361336", "end": "2018-12-05 08:35:43.453856", "rc": 0, "start": "2018-12-05 08:35:43.092520", "stderr": "", "stderr_lines": [], "stdout": "Pending", "stdout_lines": ["Pending"]}

TASK [set_fact] ****************************************************************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:152
ok: [127.0.0.1] => {"ansible_facts": {"flag": "Fail"}, "changed": false}

TASK [include_tasks] ***********************************************************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:156
included: /common/utils/update_litmus_result_resource.yml for 127.0.0.1

TASK [Generate the litmus result CR to reflect SOT (Start of Test)] ************
task path: /common/utils/update_litmus_result_resource.yml:3
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Analyze the cr yaml] *****************************************************
task path: /common/utils/update_litmus_result_resource.yml:14
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Apply the litmus result CR] **********************************************
task path: /common/utils/update_litmus_result_resource.yml:17
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Generate the litmus result CR to reflect EOT (End of Test)] **************
task path: /common/utils/update_litmus_result_resource.yml:27
changed: [127.0.0.1] => {"changed": true, "checksum": "172ad245da32aeaf259c6b1090839707494baa19", "dest": "./litmus-result.yaml", "gid": 0, "group": "root", "md5sum": "bb91bd76c32793425a40a7447b4a7257", "mode": "0644", "owner": "root", "size": 463, "src": "/root/.ansible/tmp/ansible-tmp-1543998943.73-228328549819093/source", "state": "file", "uid": 0}

TASK [Analyze the cr yaml] *****************************************************
task path: /common/utils/update_litmus_result_resource.yml:38
changed: [127.0.0.1] => {"changed": true, "cmd": "cat litmus-result.yaml", "delta": "0:00:00.279322", "end": "2018-12-05 08:35:44.872047", "rc": 0, "start": "2018-12-05 08:35:44.592725", "stderr": "", "stderr_lines": [], "stdout": "---\napiVersion: litmus.io/v1alpha1\nkind: LitmusResult\nmetadata:\n\n  # name of the litmus testcase\n  name: node-shutdown-up-with-diff-name \nspec:\n\n  # holds information on the testcase\n  testMetadata:\n    app: percona \n    chaostype: node-shutdown-up-diff-name \n\n  # holds the state of testcase,  manually updated by json merge patch\n  # result is the useful value today, but anticipate phase use in future \n  testStatus: \n    phase: completed  \n    result: Fail ", "stdout_lines": ["---", "apiVersion: litmus.io/v1alpha1", "kind: LitmusResult", "metadata:", "", "  # name of the litmus testcase", "  name: node-shutdown-up-with-diff-name ", "spec:", "", "  # holds information on the testcase", "  testMetadata:", "    app: percona ", "    chaostype: node-shutdown-up-diff-name ", "", "  # holds the state of testcase,  manually updated by json merge patch", "  # result is the useful value today, but anticipate phase use in future ", "  testStatus: ", "    phase: completed  ", "    result: Fail "]}

TASK [Apply the litmus result CR] **********************************************
task path: /common/utils/update_litmus_result_resource.yml:41
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl apply -f litmus-result.yaml", "delta": "0:00:00.419271", "end": "2018-12-05 08:35:45.431301", "failed_when_result": false, "rc": 0, "start": "2018-12-05 08:35:45.012030", "stderr": "", "stderr_lines": [], "stdout": "litmusresult.litmus.io/node-shutdown-up-with-diff-name configured", "stdout_lines": ["litmusresult.litmus.io/node-shutdown-up-with-diff-name configured"]}

TASK [include_tasks] ***********************************************************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:162
included: /common/utils/check_no_of_nodes.yml for 127.0.0.1

TASK [Waiting for application pod reschedule (evict)] **************************
task path: /common/utils/check_no_of_nodes.yml:4
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Checking the status of chaos injected node] ******************************
task path: /common/utils/check_no_of_nodes.yml:8
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Checking the no.of nodes] ************************************************
task path: /common/utils/check_no_of_nodes.yml:22
changed: [127.0.0.1] => {"attempts": 1, "changed": true, "cmd": "kubectl get nodes --no-headers | grep -vi 'master' | grep -i 'Ready'", "delta": "0:00:00.352739", "end": "2018-12-05 08:35:46.053236", "rc": 0, "start": "2018-12-05 08:35:45.700497", "stderr": "", "stderr_lines": [], "stdout": "ip-10-0-1-196.us-west-1.compute.internal   Ready   node     3h    v1.11.0\nip-10-0-1-223.us-west-1.compute.internal   Ready   node     29m   v1.11.0\nip-10-0-1-93.us-west-1.compute.internal    Ready   node     7m    v1.11.0", "stdout_lines": ["ip-10-0-1-196.us-west-1.compute.internal   Ready   node     3h    v1.11.0", "ip-10-0-1-223.us-west-1.compute.internal   Ready   node     29m   v1.11.0", "ip-10-0-1-93.us-west-1.compute.internal    Ready   node     7m    v1.11.0"]}

TASK [include_tasks] ***********************************************************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:166
included: /common/utils/check_cstor_pool_pod.yml for 127.0.0.1

TASK [Get the cstor pool pods which remains in a pending state] ****************
task path: /common/utils/check_cstor_pool_pod.yml:2
fatal: [127.0.0.1]: FAILED! => {"changed": true, "cmd": "kubectl get po -n openebs -l app=cstor-pool -o jsonpath='{.items[?(@.status.phase==\"Pending\")].metadata.name}' | cut -d '-' -f -4", "delta": "0:00:00.359822", "end": "2018-12-05 08:35:46.616208", "failed_when_result": true, "rc": 0, "start": "2018-12-05 08:35:46.256386", "stderr": "", "stderr_lines": [], "stdout": "cstor-sparse-pool-h5s8", "stdout_lines": ["cstor-sparse-pool-h5s8"]}
	to retry, use: --limit @/percona/chaos/app_node_failure_ups_diff_name/test.retry

PLAY RECAP *********************************************************************
127.0.0.1                  : ok=33   changed=17   unreachable=0    failed=2   
```

dargasudarshan commented on chaoslib/aws_chaos/chaosutil_aws.yml:
```diff
@@ -0,0 +1,11 @@
+---
+- name: Stop the AWS ec2 instances
```
Can we change the task name, as we can perform stop,start,restart,etc.. from the same util?

mittachaitu commented on apps/percona/chaos/app_node_failure_ups_diff_name/test.yml:
```diff
@@ -0,0 +1,167 @@
+            app_ns: "{{ namespace }}"
+            app_lkey: "{{ label.split('=')[0] }}"
+            app_lvalue: "{{ label.split('=')[1] }}"
+            delay: '10'
```
Based on the application we can change delay and retries instead of hardcoded in utils.

mittachaitu commented:
@ksatchit @vishnuitta @dargasudarshan @gprasath  PTAL on this PR
Note: Triggered Pipeline and it passed as expected.
Pipeline url: https://gitlab.openebs.ci/openebs/e2e-aws/-/jobs/1098

ksatchit commented on apps/percona/chaos/app_node_failure_ups_diff_name/run_litmus_test.yml:
```diff
@@ -0,0 +1,108 @@
+
+      containers:
+      - name: ansibletest
+        image: mittachaitu/ansible-runner:ci
```
openebs/ansible-runner?

vishnuitta commented:
can we change the job name  like app_node_restart_with_diff_name ?

mittachaitu commented:
@ksatchit  logs after including the util (Platform dependent AWS)
```
kubectl logs -f node-replacement-fvm5c-rzgbz -n litmus
ansible-playbook 2.7.4
  config file = None
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible-playbook
  python version = 2.7.15rc1 (default, Nov 12 2018, 14:31:15) [GCC 7.3.0]
No config file found; using defaults
/etc/ansible/hosts did not meet host_list requirements, check plugin documentation if this is unexpected
/etc/ansible/hosts did not meet script requirements, check plugin documentation if this is unexpected

PLAYBOOK: test.yml *************************************************************
1 plays in ./percona/chaos/app_node_failure_ups_diff_name/test.yml
 [WARNING]: Found variable using reserved name: action


PLAY [localhost] ***************************************************************

TASK [Gathering Facts] *********************************************************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:2
ok: [127.0.0.1]
META: ran handlers

TASK [include_tasks] ***********************************************************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:13
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [include_tasks] ***********************************************************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:17
included: /common/utils/create_testname.yml for 127.0.0.1

TASK [Record test instance/run ID] *********************************************
task path: /common/utils/create_testname.yml:3
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Construct testname appended with runID] **********************************
task path: /common/utils/create_testname.yml:7
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [include_tasks] ***********************************************************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:19
included: /chaoslib/aws_chaos/prerequisites_aws.yml for 127.0.0.1

TASK [Installing pip] **********************************************************
task path: /chaoslib/aws_chaos/prerequisites_aws.yml:2
changed: [127.0.0.1] => {"changed": true, "cmd": "apt install python-pip", "delta": "0:00:01.397367", "end": "2018-12-08 18:33:19.653497", "rc": 0, "start": "2018-12-08 18:33:18.256130", "stderr": "\nWARNING: apt does not have a stable CLI interface. Use with caution in scripts.", "stderr_lines": ["", "WARNING: apt does not have a stable CLI interface. Use with caution in scripts."], "stdout": "Reading package lists...\nBuilding dependency tree...\nReading state information...\npython-pip is already the newest version (9.0.1-2.3~ubuntu1).\n0 upgraded, 0 newly installed, 0 to remove and 16 not upgraded.", "stdout_lines": ["Reading package lists...", "Building dependency tree...", "Reading state information...", "python-pip is already the newest version (9.0.1-2.3~ubuntu1).", "0 upgraded, 0 newly installed, 0 to remove and 16 not upgraded."]}

TASK [Installing boto] *********************************************************
task path: /chaoslib/aws_chaos/prerequisites_aws.yml:7
changed: [127.0.0.1] => {"changed": true, "cmd": "pip install boto", "delta": "0:00:01.834495", "end": "2018-12-08 18:33:21.618739", "rc": 0, "start": "2018-12-08 18:33:19.784244", "stderr": "", "stderr_lines": [], "stdout": "Collecting boto\n  Downloading https://files.pythonhosted.org/packages/23/10/c0b78c27298029e4454a472a1919bde20cb182dab1662cec7f2ca1dcc523/boto-2.49.0-py2.py3-none-any.whl (1.4MB)\nInstalling collected packages: boto\nSuccessfully installed boto-2.49.0", "stdout_lines": ["Collecting boto", "  Downloading https://files.pythonhosted.org/packages/23/10/c0b78c27298029e4454a472a1919bde20cb182dab1662cec7f2ca1dcc523/boto-2.49.0-py2.py3-none-any.whl (1.4MB)", "Installing collected packages: boto", "Successfully installed boto-2.49.0"]}

TASK [Check the status of nodes and have count of nodes] ***********************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:22
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get nodes --no-headers | grep -v 'master'", "delta": "0:00:00.488112", "end": "2018-12-08 18:33:22.243534", "rc": 0, "start": "2018-12-08 18:33:21.755422", "stderr": "", "stderr_lines": [], "stdout": "ip-10-0-1-13.us-west-1.compute.internal    Ready   node     10h   v1.11.0\nip-10-0-1-232.us-west-1.compute.internal   Ready   node     10h   v1.11.0\nip-10-0-1-65.us-west-1.compute.internal    Ready   node     24m   v1.11.0", "stdout_lines": ["ip-10-0-1-13.us-west-1.compute.internal    Ready   node     10h   v1.11.0", "ip-10-0-1-232.us-west-1.compute.internal   Ready   node     10h   v1.11.0", "ip-10-0-1-65.us-west-1.compute.internal    Ready   node     24m   v1.11.0"]}

TASK [include_tasks] ***********************************************************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:29
included: /common/utils/update_litmus_result_resource.yml for 127.0.0.1

TASK [Generate the litmus result CR to reflect SOT (Start of Test)] ************
task path: /common/utils/update_litmus_result_resource.yml:3
changed: [127.0.0.1] => {"changed": true, "checksum": "6c89d31299310b5fb0f466845094d788af589c6b", "dest": "./litmus-result.yaml", "gid": 0, "group": "root", "md5sum": "dd7bb31a2659051485535d0cde60f85c", "mode": "0644", "owner": "root", "size": 450, "src": "/root/.ansible/tmp/ansible-tmp-1544294002.37-21580064841878/source", "state": "file", "uid": 0}

TASK [Analyze the cr yaml] *****************************************************
task path: /common/utils/update_litmus_result_resource.yml:14
changed: [127.0.0.1] => {"changed": true, "cmd": "cat litmus-result.yaml", "delta": "0:00:00.281137", "end": "2018-12-08 18:33:23.141316", "rc": 0, "start": "2018-12-08 18:33:22.860179", "stderr": "", "stderr_lines": [], "stdout": "---\napiVersion: litmus.io/v1alpha1\nkind: LitmusResult\nmetadata:\n\n  # name of the litmus testcase\n  name: node-replacement \nspec:\n\n  # holds information on the testcase\n  testMetadata:\n    app: percona \n    chaostype: node-shutdown-up-diff-name \n\n  # holds the state of testcase,  manually updated by json merge patch\n  # result is the useful value today, but anticipate phase use in future \n  testStatus: \n    phase: in-progress  \n    result: none ", "stdout_lines": ["---", "apiVersion: litmus.io/v1alpha1", "kind: LitmusResult", "metadata:", "", "  # name of the litmus testcase", "  name: node-replacement ", "spec:", "", "  # holds information on the testcase", "  testMetadata:", "    app: percona ", "    chaostype: node-shutdown-up-diff-name ", "", "  # holds the state of testcase,  manually updated by json merge patch", "  # result is the useful value today, but anticipate phase use in future ", "  testStatus: ", "    phase: in-progress  ", "    result: none "]}

TASK [Apply the litmus result CR] **********************************************
task path: /common/utils/update_litmus_result_resource.yml:17
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl apply -f litmus-result.yaml", "delta": "0:00:00.448848", "end": "2018-12-08 18:33:23.725749", "failed_when_result": false, "rc": 0, "start": "2018-12-08 18:33:23.276901", "stderr": "", "stderr_lines": [], "stdout": "litmusresult.litmus.io/node-replacement created", "stdout_lines": ["litmusresult.litmus.io/node-replacement created"]}

TASK [Generate the litmus result CR to reflect EOT (End of Test)] **************
task path: /common/utils/update_litmus_result_resource.yml:27
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Analyze the cr yaml] *****************************************************
task path: /common/utils/update_litmus_result_resource.yml:38
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Apply the litmus result CR] **********************************************
task path: /common/utils/update_litmus_result_resource.yml:41
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Display the app information passed via the test job] *********************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:36
ok: [127.0.0.1] => {
    "msg": [
        "The application info is as follows:", 
        "Namespace    : app-percona-ns", 
        "Label        : name=percona"
    ]
}

TASK [include_tasks] ***********************************************************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:44
included: /common/utils/status_app_pod.yml for 127.0.0.1

TASK [Checking Percona pod is in running state] ********************************
task path: /common/utils/status_app_pod.yml:2
changed: [127.0.0.1] => {"attempts": 1, "changed": true, "cmd": "kubectl get pods -n app-percona-ns -o jsonpath='{.items[?(@.metadata.labels.name==\"percona\")].status.phase}'", "delta": "0:00:00.355140", "end": "2018-12-08 18:33:24.437558", "rc": 0, "start": "2018-12-08 18:33:24.082418", "stderr": "", "stderr_lines": [], "stdout": "Running", "stdout_lines": ["Running"]}

TASK [Get application pod name] ************************************************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:54
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get po -n app-percona-ns -l name=percona --no-headers -o=custom-columns=NAME:\".metadata.name\"", "delta": "0:00:00.353195", "end": "2018-12-08 18:33:24.939496", "rc": 0, "start": "2018-12-08 18:33:24.586301", "stderr": "", "stderr_lines": [], "stdout": "percona-6966c6db86-x2xwn", "stdout_lines": ["percona-6966c6db86-x2xwn"]}

TASK [Generate some random name for database creation] *************************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:62
changed: [127.0.0.1] => {"changed": true, "cmd": "echo $(mktemp) | cut -d '.' -f 2", "delta": "0:00:00.282810", "end": "2018-12-08 18:33:25.355335", "rc": 0, "start": "2018-12-08 18:33:25.072525", "stderr": "", "stderr_lines": [], "stdout": "vbhzUtamIt", "stdout_lines": ["vbhzUtamIt"]}

TASK [Create test load in the mysql database] **********************************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:70
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Get Application pod Node to perform chaos] *******************************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:82
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get pod percona-6966c6db86-x2xwn -n app-percona-ns --no-headers -o custom-columns=:spec.nodeName", "delta": "0:00:00.355870", "end": "2018-12-08 18:33:25.890280", "rc": 0, "start": "2018-12-08 18:33:25.534410", "stderr": "", "stderr_lines": [], "stdout": "ip-10-0-1-13.us-west-1.compute.internal", "stdout_lines": ["ip-10-0-1-13.us-west-1.compute.internal"]}

TASK [include_tasks] ***********************************************************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:91
included: /common/utils/fetch_aws_details.yml for 127.0.0.1

TASK [Get the AWS instance Id of APP NOde] *************************************
task path: /common/utils/fetch_aws_details.yml:3
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get node ip-10-0-1-13.us-west-1.compute.internal -o jsonpath={.spec.providerID} | cut -d '/' -f 5", "delta": "0:00:00.352628", "end": "2018-12-08 18:33:26.442100", "rc": 0, "start": "2018-12-08 18:33:26.089472", "stderr": "", "stderr_lines": [], "stdout": "i-0578abec273c95f73", "stdout_lines": ["i-0578abec273c95f73"]}

TASK [Get region from app node] ************************************************
task path: /common/utils/fetch_aws_details.yml:9
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get nodes ip-10-0-1-13.us-west-1.compute.internal -o jsonpath='{.metadata.labels.failure-domain\\.beta\\.kubernetes\\.io\\/region}'", "delta": "0:00:00.354543", "end": "2018-12-08 18:33:26.934060", "rc": 0, "start": "2018-12-08 18:33:26.579517", "stderr": "", "stderr_lines": [], "stdout": "us-west-1", "stdout_lines": ["us-west-1"]}

TASK [set_fact] ****************************************************************
task path: /common/utils/fetch_aws_details.yml:16
ok: [127.0.0.1] => {"ansible_facts": {"instance_id": "i-0578abec273c95f73", "region": "us-west-1"}, "changed": false}

TASK [include_tasks] ***********************************************************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:95
included: /chaoslib/aws_chaos/chaosutil_aws.yml for 127.0.0.1

TASK [Performing action on AWS ec2 instances] **********************************
task path: /chaoslib/aws_chaos/chaosutil_aws.yml:2
changed: [127.0.0.1] => {"changed": true, "instance_ids": ["i-0578abec273c95f73"], "instances": [{"ami_launch_index": "0", "architecture": "x86_64", "block_device_mapping": {"/dev/sda1": {"delete_on_termination": true, "status": "attached", "volume_id": "vol-026d66c141337bf59"}}, "dns_name": "", "ebs_optimized": false, "groups": {"sg-03a4da4ce3ead292b": "nodes.k8s-sai-ubuntu.k8s.local"}, "hypervisor": "xen", "id": "i-0578abec273c95f73", "image_id": "ami-0ad16744583f21877", "instance_type": "t2.xlarge", "kernel": null, "key_name": "kubernetes.k8s-sai-ubuntu.k8s.local-14:ef:42:fe:eb:1c:4e:6f:1f:f1:e1:87:75:95:96:df", "launch_time": "2018-12-08T07:46:18.000Z", "placement": "us-west-1a", "private_dns_name": "ip-10-0-1-13.us-west-1.compute.internal", "private_ip": "10.0.1.13", "public_dns_name": "", "public_ip": null, "ramdisk": null, "region": "us-west-1", "root_device_name": "/dev/sda1", "root_device_type": "ebs", "state": "stopped", "state_code": 80, "tags": {"KubernetesCluster": "k8s-sai-ubuntu.k8s.local", "Name": "nodes.k8s-sai-ubuntu.k8s.local", "aws:autoscaling:groupName": "nodes.k8s-sai-ubuntu.k8s.local", "k8s.io/cluster-autoscaler/node-template/label/kops.k8s.io/instancegroup": "nodes", "k8s.io/role/node": "1"}, "tenancy": "default", "virtualization_type": "hvm"}], "tagged_instances": []}

TASK [include_tasks] ***********************************************************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:98
included: /common/utils/check_no_of_nodes.yml for 127.0.0.1

TASK [Waiting for application pod reschedule (evict)] **************************
task path: /common/utils/check_no_of_nodes.yml:4
ok: [127.0.0.1] => {"changed": false, "elapsed": 50, "path": null, "port": null, "search_regex": null, "state": "started"}

TASK [Checking the status of chaos injected node] ******************************
task path: /common/utils/check_no_of_nodes.yml:8
 [WARNING]: when statements should not include jinja2 templating delimiters
such as {{ }} or {% %}. Found: '{{ app_node.stdout }}' not in node_names.stdout

changed: [127.0.0.1] => {"attempts": 1, "changed": true, "cmd": "kubectl get nodes --no-headers | grep -vi 'master'", "delta": "0:00:00.355353", "end": "2018-12-08 18:34:49.323509", "rc": 0, "start": "2018-12-08 18:34:48.968156", "stderr": "", "stderr_lines": [], "stdout": "ip-10-0-1-232.us-west-1.compute.internal   Ready   node     10h   v1.11.0\nip-10-0-1-65.us-west-1.compute.internal    Ready   node     25m   v1.11.0", "stdout_lines": ["ip-10-0-1-232.us-west-1.compute.internal   Ready   node     10h   v1.11.0", "ip-10-0-1-65.us-west-1.compute.internal    Ready   node     25m   v1.11.0"]}

TASK [Checking the no.of nodes] ************************************************
task path: /common/utils/check_no_of_nodes.yml:22
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [include_tasks] ***********************************************************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:103
included: /common/utils/status_app_pod.yml for 127.0.0.1

TASK [Checking Percona post chaos pod is in running state] *********************
task path: /common/utils/status_app_pod.yml:2
FAILED - RETRYING: Checking Percona post chaos pod is in running state (30 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (29 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (28 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (27 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (26 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (25 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (24 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (23 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (22 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (21 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (20 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (19 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (18 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (17 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (16 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (15 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (14 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (13 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (12 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (11 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (10 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (9 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (8 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (7 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (6 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (5 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (4 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (3 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (2 retries left).
FAILED - RETRYING: Checking Percona post chaos pod is in running state (1 retries left).
fatal: [127.0.0.1]: FAILED! => {"attempts": 30, "changed": true, "cmd": "kubectl get pods -n app-percona-ns -o jsonpath='{.items[?(@.metadata.labels.name==\"percona\")].status.phase}'", "delta": "0:00:00.350688", "end": "2018-12-08 18:40:03.404411", "rc": 0, "start": "2018-12-08 18:40:03.053723", "stderr": "", "stderr_lines": [], "stdout": "Pending", "stdout_lines": ["Pending"]}

TASK [set_fact] ****************************************************************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:139
ok: [127.0.0.1] => {"ansible_facts": {"flag": "Fail"}, "changed": false}

TASK [include_tasks] ***********************************************************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:143
included: /common/utils/update_litmus_result_resource.yml for 127.0.0.1

TASK [Generate the litmus result CR to reflect SOT (Start of Test)] ************
task path: /common/utils/update_litmus_result_resource.yml:3
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Analyze the cr yaml] *****************************************************
task path: /common/utils/update_litmus_result_resource.yml:14
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Apply the litmus result CR] **********************************************
task path: /common/utils/update_litmus_result_resource.yml:17
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Generate the litmus result CR to reflect EOT (End of Test)] **************
task path: /common/utils/update_litmus_result_resource.yml:27
changed: [127.0.0.1] => {"changed": true, "checksum": "d168e1bba34a32789adf3b7ea22bad77e0b426db", "dest": "./litmus-result.yaml", "gid": 0, "group": "root", "md5sum": "c2e52ecf7226fff51117c3d6b9ff9ad7", "mode": "0644", "owner": "root", "size": 448, "src": "/root/.ansible/tmp/ansible-tmp-1544294403.68-257087846009718/source", "state": "file", "uid": 0}

TASK [Analyze the cr yaml] *****************************************************
task path: /common/utils/update_litmus_result_resource.yml:38
changed: [127.0.0.1] => {"changed": true, "cmd": "cat litmus-result.yaml", "delta": "0:00:00.282641", "end": "2018-12-08 18:40:04.825658", "rc": 0, "start": "2018-12-08 18:40:04.543017", "stderr": "", "stderr_lines": [], "stdout": "---\napiVersion: litmus.io/v1alpha1\nkind: LitmusResult\nmetadata:\n\n  # name of the litmus testcase\n  name: node-replacement \nspec:\n\n  # holds information on the testcase\n  testMetadata:\n    app: percona \n    chaostype: node-shutdown-up-diff-name \n\n  # holds the state of testcase,  manually updated by json merge patch\n  # result is the useful value today, but anticipate phase use in future \n  testStatus: \n    phase: completed  \n    result: Fail ", "stdout_lines": ["---", "apiVersion: litmus.io/v1alpha1", "kind: LitmusResult", "metadata:", "", "  # name of the litmus testcase", "  name: node-replacement ", "spec:", "", "  # holds information on the testcase", "  testMetadata:", "    app: percona ", "    chaostype: node-shutdown-up-diff-name ", "", "  # holds the state of testcase,  manually updated by json merge patch", "  # result is the useful value today, but anticipate phase use in future ", "  testStatus: ", "    phase: completed  ", "    result: Fail "]}

TASK [Apply the litmus result CR] **********************************************
task path: /common/utils/update_litmus_result_resource.yml:41
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl apply -f litmus-result.yaml", "delta": "0:00:00.420960", "end": "2018-12-08 18:40:05.382500", "failed_when_result": false, "rc": 0, "start": "2018-12-08 18:40:04.961540", "stderr": "", "stderr_lines": [], "stdout": "litmusresult.litmus.io/node-replacement configured", "stdout_lines": ["litmusresult.litmus.io/node-replacement configured"]}

TASK [include_tasks] ***********************************************************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:149
included: /common/utils/check_no_of_nodes.yml for 127.0.0.1

TASK [Waiting for application pod reschedule (evict)] **************************
task path: /common/utils/check_no_of_nodes.yml:4
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Checking the status of chaos injected node] ******************************
task path: /common/utils/check_no_of_nodes.yml:8
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Checking the no.of nodes] ************************************************
task path: /common/utils/check_no_of_nodes.yml:22
changed: [127.0.0.1] => {"attempts": 1, "changed": true, "cmd": "kubectl get nodes --no-headers | grep -vi 'master' | grep -i 'Ready'", "delta": "0:00:00.353358", "end": "2018-12-08 18:40:06.006999", "rc": 0, "start": "2018-12-08 18:40:05.653641", "stderr": "", "stderr_lines": [], "stdout": "ip-10-0-1-232.us-west-1.compute.internal   Ready   node     10h   v1.11.0\nip-10-0-1-33.us-west-1.compute.internal    Ready   node     3m    v1.11.0\nip-10-0-1-65.us-west-1.compute.internal    Ready   node     30m   v1.11.0", "stdout_lines": ["ip-10-0-1-232.us-west-1.compute.internal   Ready   node     10h   v1.11.0", "ip-10-0-1-33.us-west-1.compute.internal    Ready   node     3m    v1.11.0", "ip-10-0-1-65.us-west-1.compute.internal    Ready   node     30m   v1.11.0"]}

TASK [include_tasks] ***********************************************************
task path: /percona/chaos/app_node_failure_ups_diff_name/test.yml:153
included: /common/utils/check_cstor_pool_pod.yml for 127.0.0.1

TASK [Get the cstor pool pods which remains in a pending state] ****************
task path: /common/utils/check_cstor_pool_pod.yml:2
fatal: [127.0.0.1]: FAILED! => {"changed": true, "cmd": "kubectl get po -n openebs -l app=cstor-pool -o jsonpath='{.items[?(@.status.phase==\"Pending\")].metadata.name}' | wc -l", "delta": "0:00:00.355772", "end": "2018-12-08 18:40:06.575589", "failed_when_result": true, "rc": 0, "start": "2018-12-08 18:40:06.219817", "stderr": "", "stderr_lines": [], "stdout": "0", "stdout_lines": ["0"]}
	to retry, use: --limit @/percona/chaos/app_node_failure_ups_diff_name/test.retry

PLAY RECAP *********************************************************************
127.0.0.1                  : ok=34   changed=18   unreachable=0    failed=2  
```

ksatchit commented on common/utils/check_cstor_pool_pod.yml:
```diff
@@ -0,0 +1,7 @@
+         kubectl get po -n {{ pool_namespace }} -l {{ pool_label }} 
+         -o jsonpath='{.items[?(@.status.phase=="Pending")].metadata.name}' | wc -w
+     register: pod_pending
+     failed_when: "pod_pending.stdout != 0"
```
can take this now, but really recommend to check if this is a subset of other utils. Lets try to avoid redundancy in utils.

ksatchit commented on common/utils/check_no_of_nodes.yml:
```diff
@@ -0,0 +1,31 @@
+     retries: 10
+     delay: 60
+
+  when: injected_node_chaos == "no"
```
Looks like a case of test business logic flowing into a util ? For ex: util to check (Ready) node count is great. But `waits`, actions for `pre` & `post` chaos may still be in test. However, this is just an opinion.

mittachaitu replied:
Modified the condition to check (Ready) node count.

ksatchit commented on common/utils/fetch_aws_details.yml:
```diff
@@ -0,0 +1,18 @@
+---
```
Maybe can go in AWS folder inside utils ?

ksatchit commented:
Good one @mittachaitu ! !
